### PR TITLE
Activity Log: Step 1 for design v1

### PR DESCRIPTION
**Pull request by elibud**
Another step on https://github.com/wordpress-mobile/WordPress-iOS/issues/7832

- Updated Activity icon on the blog settings screen.
- Grouped activities by date on the feed.
- Filtered out discarded activities.

**To test:**

On a site with AL enabled check that what  you see on mobile matches the web minus the discarded activities, which will not be displayed on mobile for now as per the last designs from @mattmiklic.

The grouping code on `ActivityListViewModel` is not meant to be permanent since we will have to face the paging + persistence issue eventually.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Activity/ActivityListViewModel.swift [hunk 1]
```diff
@@ -40,17 +40,41 @@ enum ActivityListViewModel {
...
                         presenter.presentRewindFor(activity: activity)
                     }
                 )
             })
-            return ImmuTable(sections: [
-                ImmuTableSection(rows: rows)
-            ])
+
+            guard rows.count > 0 else {
+                return ImmuTable(sections: [])
```
Comment (koke) [inline WordPress/Classes/ViewRelated/Activity/ActivityListViewModel.swift]: This is fine, but just FYI, you can use the shorthand `ImmuTable.Empty` for this

File: WordPress/Classes/ViewRelated/Activity/ActivityListViewModel.swift [hunk 2]
```diff
@@ -40,17 +40,41 @@ enum ActivityListViewModel {
...
+                                                          optionalRows: currentSection.rows,
+                                                          footerText: nil)!)
+                    currentSection = (row.activity.published.longUTCStringWithoutTime(), [row])
+                }
+            }
+            tableSections.append(ImmuTableSection(headerText: currentSection.date,
+                                                  optionalRows: currentSection.rows,
+                                                  footerText: nil)!)
+
+            return ImmuTable(sections: tableSections)
```
Comment (koke) [inline WordPress/Classes/ViewRelated/Activity/ActivityListViewModel.swift]: This was a bit hard to follow. I'd try to move the algorithm for grouping somewhere else, maybe an `Array` extension that returns a `[String: [Element]]`.

I have to go for today and I don't have a prettier algorithm, but here's a stab at a generic version that can be unit tested:

```swift
extension Array {
    func sortedGroup<Value: Equatable>(keyPath: KeyPath<Element, Value>) -> [(Value, [Element])] {
        var currentValue: Value?
        var currentGroup = [Element]()
        var result = [(Value, [Element])]()
        forEach { (element) in
            let value = element[keyPath: keyPath]
            if currentValue != value {
                if let currentValue = currentValue {
                    result.append((currentValue, currentGroup))
                }
                currentValue = value
                currentGroup = []
            }
            currentGroup.append(element)
        }
        if let currentValue = currentValue {
            result.append((currentValue, currentGroup))
        }
        return result
    }
}
```

In any case, you can probably use `ImmuTable(optionalSections:)` and avoid those `!`

Comment (koke) [reply WordPress/Classes/ViewRelated/Activity/ActivityListViewModel.swift]: Actually, I don't thin there's a need for the `optionalRows` variant of `ImmuTableSection`. The `currentSection.rows` is an array of non-optional rows, so you can use the regular section.

Comment (elibud) [reply WordPress/Classes/ViewRelated/Activity/ActivityListViewModel.swift]: 😍 will do!

File: WordPress/Classes/ViewRelated/Activity/ActivityListViewModel.swift (hunk 1 above)
Comment (elibud) [reply WordPress/Classes/ViewRelated/Activity/ActivityListViewModel.swift]: 👍 good to know!

**Comment by elibud** [comment]
@koke thanks for the suggestions, it looks much better now. Ready for another 👀

File: WordPress/Classes/ViewRelated/Activity/ActivityListViewModel.swift [hunk 3]
```diff
@@ -40,17 +40,27 @@ enum ActivityListViewModel {
...
+            let publishedDateUTCKeyPath = \ActivityListRow.activity.publishedDateUTCWithoutTime
+            let groupedRows = rows.sortedGroup(keyPath: publishedDateUTCKeyPath)
+
+            let tableSections = groupedRows.map { (date, rows) in
+                return ImmuTableSection(headerText: date,
+                                        optionalRows: rows,
+                                        footerText: nil)
+            }
+
+            return ImmuTable(optionalSections: tableSections)
```
Comment (koke) [inline WordPress/Classes/ViewRelated/Activity/ActivityListViewModel.swift]: I don't think you'd get any `nil` rows coming from `sortedGroup`, but the compiler doesn't know that, so I guess the `optional` ImmuTable variants are 👌 here
